### PR TITLE
docs: add `v1.2.0` release details to the CHANGELOG

### PR DESCRIPTION
**Pull request by jon-whit**
## Description
Add the `v1.2.0` release details to the CHANGELOG in preparation for the upcoming release.

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [x] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

**Comment by jon-whit** [comment]
Will merge https://github.com/openfga/api/pull/80 after we release `v1.2.0`, which will be right after we merge this PR.

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,16 @@ Try to keep listed changes to a concise bulleted list of simple explanations of
 
 ## [Unreleased]
 
+## [1.2.0] - 2023-07-01
```
Comment (rhamzeh) [inline CHANGELOG.md:11]:
```suggestion
## [1.2.0] - 2023-06-30
```

**Review by rhamzeh — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (jon-whit) [reply CHANGELOG.md:11]: Oops 😄

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,16 @@ Try to keep listed changes to a concise bulleted list of simple explanations of
...
 
+## [1.2.0] - 2023-06-30
+
+[Full changelog](https://github.com/openfga/openfga/compare/v1.1.1...v1.2.0)
+
+### Added
+* Optimizations for [ListObjects](https://openfga.dev/api/service#/Relationship%20Queries/ListObjects) and [StreamedListObjects](https://openfga.dev/api/service#/Relationship%20Queries/StreamedListObjects) for models involving intersection (`and`) and exclusion (`but not`) (#797)
+
+### Changed
+* Memoize typesystem resolution and model validation instead of requiring re-evaluation (#831)
```
Comment (adriantam) [inline CHANGELOG.md:19]: Should we mention something about improving performance ?  From end user perspective, they will have no idea how it affects them

Comment (miparnisari) [reply CHANGELOG.md:19]: We can say that now we cache the latest authorization model in between requests

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,16 @@ Try to keep listed changes to a concise bulleted list of simple explanations of
...
 ## [Unreleased]
 
+## [1.2.0] - 2023-06-30
+
+[Full changelog](https://github.com/openfga/openfga/compare/v1.1.1...v1.2.0)
+
+### Added
+* Optimizations for [ListObjects](https://openfga.dev/api/service#/Relationship%20Queries/ListObjects) and [StreamedListObjects](https://openfga.dev/api/service#/Relationship%20Queries/StreamedListObjects) for models involving intersection (`and`) and exclusion (`but not`) (#797)
+
+### Changed
```
Comment (miparnisari) [inline CHANGELOG.md:18]: can you add https://github.com/openfga/openfga/pull/829

i'd like people that are using this sample to re-download it

**Review by adriantam — APPROVED** [review]

**Review by rhamzeh — DISMISSED** [review]

**Review by rhamzeh — APPROVED** [review]

**Review by miparnisari — APPROVED** [review]

**Pull request merged**